### DISCUSSION TITLE
Where is the database schema?

### DISCUSSION THREAD

**Discussion by ghost** [beetbox/beets#4355] [Q&A] [answered]
I only saw a reference to _models in dbcore.

**Comment by jackwilsdon** [comment, marked as answer]
The schema is dynamically generated at runtime by `Database` using the models configured on `Library`:
https://github.com/beetbox/beets/blob/e06cf7969bfdfa4773049699320471be45d56054/beets/dbcore/db.py#L926-L929
https://github.com/beetbox/beets/blob/e06cf7969bfdfa4773049699320471be45d56054/beets/library.py#L1448-L1450

The two models configured are `Item` and `Album`:
https://github.com/beetbox/beets/blob/e06cf7969bfdfa4773049699320471be45d56054/beets/library.py#L456-L543
https://github.com/beetbox/beets/blob/e06cf7969bfdfa4773049699320471be45d56054/beets/library.py#L1033-L1081
